Layout: Add anchor link affordance to all pages

harding opened the pull request:
This pull calls a Javascript function to display a hyperlinked icon whenever a Javascript-enabled user hovers on any subhead on the site that has an anchor set.  Combined with PR #813, it makes it easy for users to link to any section within the site.

This pull request is being added to the bitcoin-core-0.10.1 branch because updating the Javascript will trigger an alert to people monitoring the page.  Since that alert will be triggered when 0.10.1 is released, this will avoid an unnecessary extra alert.

Live preview of this PR plus PR #813 merged together: http://dg1.dtrt.org/en/bitcoin-for-individuals

Screenshots of the anchor link icon in different layouts:

![2015-04-09-112414_1280x800_scrot](https://cloud.githubusercontent.com/assets/61096/7070841/8c1ed182-deaf-11e4-8f49-8c1f9a4a93c7.png)

![2015-04-09-112425_1280x800_scrot](https://cloud.githubusercontent.com/assets/61096/7070846/933023b8-deaf-11e4-843e-b71bf362240f.png)

Note: this branch will fail Travis CI because its based on the 0.10.1 branch that itself fails until 0.10.1 final is uploaded to the site.

saivann commented:
@harding Looks good! One bug: the "addAnchorLinks" function should be moved to base.js otherwise it will be called on the download page, and fail since the function won't exist on that page. I don't think this function calls other functions that are missing in base.js, but it's probably worth checking in your javascript console to confirm there isn't any error.

harding commented:
@saivann good catch.  Thanks!  In this branch (pull #814), I moved the JS code as you suggested.  In the other branch (pull #813) I fixed a stylesheet issue that made the link icon appear all the way on the side of the page even though the header text is centered here.  For that, I just copied @buren's method from a couple days ago.

I've updated the combined preview and verified there are no JS errors: http://dg1.dtrt.org/en/download
